airflow/config_templates/default_airflow.cfg

```diff
@@ -208,6 +208,8 @@ dag_discovery_safe_mode = True
 # The number of retries each task is going to have by default. Can be overridden at dag or task level.
 default_task_retries = 0
 
+check_task_pools = False
```

Comment:
Why this as an option? I think we should check if the pool exists anyway. Currently, the error message is horrible :-)

Reply:
I was thinking it would technically be a breaking change to fail dag parsing on missing pools.  What do you think? cc @ashb @potiuk, since you all discussed this issue in https://github.com/apache/airflow/pull/6451.